src/hacbs/components/ApplicationDetails/tabs/overview/visualization/hooks/useAppWorkflowData.ts

```diff
@@ -1,129 +1,582 @@
+    [applicationTestNodes, expanded],
+  );
+
+  const testsGroup = React.useMemo(
+    () =>
+      allResourcesLoaded
+        ? groupToPipelineNode(
+            'tests',
+            'Tests',
+            WorkflowNodeType.TESTS,
+            buildTasks,
+            expanded,
+            expanded
+              ? [...componentIntegrationTestTasks, ...applicationIntegrationTestTasks]
+              : undefined,
+            [...componentTestNodes, ...applicationTestNodes],
+            [...componentIntegrationTests, ...applicationIntegrationTests],
+          )
+        : undefined,
+    [
+      allResourcesLoaded,
+      applicationIntegrationTestTasks,
+      applicationIntegrationTests,
+      applicationTestNodes,
+      buildTasks,
+      componentIntegrationTestTasks,
+      componentIntegrationTests,
+      componentTestNodes,
+      expanded,
+    ],
+  );
 
-  const nodes: WorkflowNode[] = workflowToNodes(workflowObject);
+  const staticEnvironments = environments.filter(
+    (e) => getEnvironmentType(e) === EnvironmentType.static,
+  );
+
+  const staticEnvironmentNodes = React.useMemo(() => {
+    const nodes = staticEnvironments.length
+      ? staticEnvironments.map((staticEnvironment) => {
+          const prevEnv = staticEnvironments.find(
+            (e) => e.metadata.name === staticEnvironment.spec.parentEnvironment,
+          );
+          return resourceToPipelineNode(
+            staticEnvironment,
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            prevEnv ? [prevEnv.metadata.uid] : applicationIntegrationTestTasks,
+            healthStatus,
+          );
+        })
+      : [
+          emptyPipelineNode(
+            'no-static-environments',
+            'Static environments',
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            applicationIntegrationTestTasks,
+          ),
+        ];
+    const maxWidth = nodes.reduce((max, node) => Math.max(max, node.width), 0);
+    nodes.forEach((n) => (n.width = maxWidth));
+    return nodes;
+  }, [applicationIntegrationTestTasks, healthStatus, staticEnvironments]);
+
+  const staticEnvironmentGroup = React.useMemo(
+    () =>
+      allResourcesLoaded && testsGroup?.id
+        ? groupToPipelineNode(
+            'static-environments',
+            'Static environments',
+            WorkflowNodeType.STATIC_ENVIRONMENT,
+            expanded ? applicationIntegrationTestTasks : [testsGroup.id],
+            expanded,
+            expanded ? staticEnvironmentNodes?.map((c) => c.id) : undefined,
+            staticEnvironmentNodes,
+            staticEnvironments,
+            healthStatus,
+          )
+        : undefined,
+    [
+      allResourcesLoaded,
+      applicationIntegrationTestTasks,
+      expanded,
+      healthStatus,
+      staticEnvironmentNodes,
+      staticEnvironments,
+      testsGroup?.id,
+    ],
+  );
+
+  const lastStaticEnv = getLastEnvironments(staticEnvironments);
```

Comment:
You need to use `useMemo` here since `lastStaticEnv` is used as a memo dependency. Otherwise the dependency check will always fail.